railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -18,7 +18,7 @@ def build_stack
           middleware.use ::Rack::Sendfile, config.action_dispatch.x_sendfile_header
 
           if config.serve_static_files
-            middleware.use ::ActionDispatch::Static, paths["public"].first, config.static_cache_control
+            middleware.use ::ActionDispatch::Static, paths["public"].first, config.static_cache_control, config.action_dispatch.asset_headers
           end
```

Comment:
Added a deprecation in the setter, I'm not sure if it looks good enough, though. I'll send a separate pull request for replacing `config.serve_static_files` with `config.public_file_server.enabled`.